crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -365,7 +374,11 @@ async fn get_vectors_with_chunker(
         values.append_slice(&inner); // I believe this is a clone under the hood.
 
         // Get the length of the last chunk
-        let last_chunk_length = embedded_data.as_slice()[curr + chunks_in_row - 1].len();
```

Comment:
if `curr + chunks_in_row == 0` this `[curr + chunks_in_row - 1]` will be -18446744073709551615 and an index out of range.